jetty-core/jetty-deploy/src/main/java/org/eclipse/jetty/deploy/ContextHandlerDeployer.java

```diff
@@ -0,0 +1,64 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.deploy;
+
+import org.eclipse.jetty.server.handler.ContextHandler;
+
+/**
+ * Deployer for ContextHandlers
+ */
+public interface ContextHandlerDeployer
+{
+    /**
+     * Add a ContextHandler to the graph and then move it a deployed state.
+     *
+     * @param contextHandler the ContextHandler to deploy.
+     */
+    void deploy(ContextHandler contextHandler);
+
+    /**
+     * Move a ContextHandler int the graph to an undeployed state, and then remove it from the graph.
+     *
+     * @param contextHandler the ContextHandler to undeploy.
+     */
+    void undeploy(ContextHandler contextHandler);
+
+    /**
+     * Add a ContextHandler int the graph but perform no actions on it, leaving it in undeployed state.
+     *
+     * @param contextHandler the ContextHandler to add to the graph.
+     */
+    default void addUndeployed(ContextHandler contextHandler)
+    {
+    }
+
+    /**
+     * Advanced usage, move a ContextHandler through the DeploymentManager graph by name.
+     *
+     * @param contextHandler the ContextHandler to move
+     * @param goalName the goal graph node by name
+     */
+    default void move(ContextHandler contextHandler, String goalName)
+    {
+    }
```

Comment:
Would this be better?  It would allow alternate ContextHandlerDeployers to return null to indicate they do not support goals.

```suggestion
    /**
     * Create a goal driven Deployer for a ContextHandler 
     * @param contextHandler the ContextHandler .
     */
    default GoalDriven createGoalDrivenDeployer(ContextHandler contextHandler)
    {
        return null;
    }

    public interface GoalDriven
    {
        ContextHandler getContextHandler();
        
        String getCurrentGoal()
        
        /**
         * Advanced usage, move a ContextHandler through the DeploymentManager graph by name.
         *
         * @param goalName the goal graph node by name
         */
        default void move(String goalName)
        {
        }
    }
```

Reply:
No, this is overly complicated and not useful.

Reply:
If you want to iterate this kind of thing, lets do it AFTER the fundamentals of this PR are merged.

Reply:
Now is the time to get the APIs right. It does have benifits as it clearly demarcates the usage of the API in default mode vs in goal oriented mode. It prevents the move API being used on contexts that were not added undeployed.